### ISSUE TITLE
Functions are not granted access to notifications

### ISSUE THREAD

**Issue by struct78** [aws-amplify/amplify-cli#5157]
**Describe the bug**
Updating a lambda function to give it access to a notifications resource does not update any of the functions parameters.

**Amplify CLI Version**
4.27.3

**To Reproduce**
1. Add notifications using `amplify add notifications`, use all the defaults (I just chose APNS)
2. Add a lambda function using `amplify add function`, use all the default options
3. When you are prompted with "Do you want to access other resources in this project from your Lambda function?", select notifications
4. When the function is generated:
a) The cloud formation template contains no execution policy to access the notifications service
b) The functions parameters aren't updated with the projectId of the notifications service

The same process using `amplify function update' does not work either.

**Expected behavior**
I would expect the projectId to be available as an environment variable within the function so that I can send push notifications, and I would expect the cloudformation template to be updated to grant me access to the notifications service.

Instead I have to manually add this to the function's cloudformation file:

```
{
	"Effect": "Allow",
	"Action": [
		"mobiletargeting:SendMessages"
	],
	"Resource": "*"
}
```

Then manually add a parameter and environment variable called `pinpointProjectId` and PINPOINT_PROJECT_ID respectively to the cloudformation file:

```
"pinpointProjectId": {
    "Type": "String",
    "Default": "pinpointProjectId"
}
```

```
"Environment": {
	"Variables": {
		"ENV": {
			"Ref": "env"
		},
		"REGION": {
			"Ref": "AWS::Region"
		},
		"PINPOINT_PROJECT_ID": {
			"Ref": "pinpointProjectId"
		}
	}
}
```

I then update the value of that environment variable manually in the Lambda console for each environment.

**Desktop (please complete the following information):**
 - OS: Mac OS 10.15.5
 - Node Version: 10.16.0

**Comment by UnleashedMind** [comment]
This is known, and the Amplify CLI prints out the the message: "Policies cannot be added for notifications/<resourcename>". 
We will add the implementation in the notifications category plugin so it exports access policy statements.

**Comment by migdonio1** [comment]
Did you try enabling the `analytics` feature? In my case this add an `ANALYTICS_EWAPINPOINT_ID` and `ANALYTICS_EWAPINPOINT_REGION` environment variables and I was able to send notifications using APNS

```js
/* Amplify Params - DO NOT EDIT
	ANALYTICS_DEMOPINPOINT_ID
	ANALYTICS_DEMOPINPOINT_REGION
        ...
Amplify Params - DO NOT EDIT */

const pinpointApplicationId = process.env.ANALYTICS_DEMOPINPOINT_ID

```

**Comment by struct78** [comment]
@migdonio1 Thanks for the tip! I tried it last night, and it did not push either of those values to the Lambda's environment variables, which causes the push to fail (as it is using FN::join to concatenate the name of the resource)

I'll just stick with my manual work around for the moment.

**Comment by vitalbone** [comment]
@renebrandel How did you go with this issue in the [Bug Bash](https://github.com/aws-amplify/amplify-cli/projects/7)? I can't seem to find a reference to it in that project.

Would be good to label this as a bug rather than an enhancement since it makes the notifications category unusable unless adding config manually.

**Comment by avlonder** [comment]
I get the error: *Error: Method getPermissionPolicies does not exist or not a function in category plugin: notifications.*

**Comment by edwardfoyle** [comment]
This is resolved in version 10.8.0 of the CLI. Please upgrade to try it out.
